internal/utils/path/path.go

```diff
@@ -57,6 +57,33 @@ func ListDirsAndFiles(paths []string) (dirs sets.Set[string], files sets.Set[str
 	return
 }
 
+// Traverses the directory recursively and adds the same to the subDirs.
+func traverseSubDirs(curDir string, subDirs sets.Set[string]) {
+	subDirs.Insert(curDir)
+	files, err := os.ReadDir(curDir)
+	if err != nil {
+		// skip
+		return
+	}
+
+	for _, file := range files {
+		if file.IsDir() {
+			fpath := filepath.Join(curDir, file.Name())
+			traverseSubDirs(fpath, subDirs)
+		}
+	}
+}
+
+// GetSubDirs returns all the subdirectories of given directories.
+// The passed directories are also included in the result.
+func GetSubDirs(initDirs []string) sets.Set[string] {
```

Comment:
not a fan of the naming of this parameter `initDirs`, it has no meaning/context in this helper function, prefer `inputDirs` or something else.

Reply:
Got it! Thanks.